index.js

```diff
@@ -1,90 +1,73 @@
-allowChecker = function(user) {
-  return true;
-}
-var allowedUsers = [];
-if(process.env.ALLOWED_USERS) {
-  allowedUsers = process.env.ALLOWED_USERS.split(',');
-  allowChecker = function(userId) {
-    var user = rtm.dataStore.getUserById(userId);
-    if(allowedUsers.lastIndexOf(user.name) != -1) {
-      return true;
-    }
-    return false;
+const allowChecker = userId => {
+  if (process.env.ALLOWED_USERS) {
+    const allowedUsers = process.env.ALLOWED_USERS.split(',')
+    const user = rtm.dataStore.getUserById(userId)
+    if (allowedUsers.lastIndexOf(user.name) !== -1) return true
+    return false
   }
+  return true
 }
 
 // The client will emit an RTM.AUTHENTICATED event on successful connection, with the `rtm.start` payload if you want to cache it
-rtm.on(CLIENT_EVENTS.RTM.AUTHENTICATED, function (rtmStartData) {
-  console.log(`Logged in as ${rtmStartData.self.name} of team ${rtmStartData.team.name}, but not yet connected to a channel`);
-});
-
-var pty = require('node-pty');
+rtm.on(Slack.CLIENT_EVENTS.RTM.AUTHENTICATED, rtmStartData => {
+  console.log(`Logged in as ${rtmStartData.self.name} of team ${rtmStartData.team.name}.`)
+  console.log('Waiting for commands..')
+})
 
 // Emitted when there's a message on a channel where the bot is participating
-rtm.on(RTM_EVENTS.MESSAGE, function handleRtmMessage(message) {
+rtm.on(Slack.RTM_EVENTS.MESSAGE, message => {
   try {
-    if(message.text.lastIndexOf(botKeyword, 0) === 0) {
-      if (allowChecker(message.user)) {
-        cmd = "kontena " + message.text.substring(botKeyword.length);
-        var term = pty.spawn('kontena', message.text.substring(botKeyword.length).trim().split(' '), {
-          name: 'xterm-color',
-          cols: 200,
-          rows: 50,
-          cwd: process.env.HOME,
-          env: process.env
-        });
-        // Send typing notification to indicate that the work is ongoing
-        const intervalObj = setInterval(() => {
-          rtm.sendTyping(message.channel);
-        }, 2000);
-        // Collect all output data to an array
-        output = []
-        term.on('data', function(data) {
-          // Strip out color codes etc.
-          out = stripAnsi(data)
-          output.push(out);
-        });
-        // When the spawned process exits
-        term.on('exit', function(code, signal) {
-          // Strip out output rows that look like plain spinner
-          msg = output.filter(function(e) {
-            return !/^[\b|\r]+[\s\S]*[\\|\|\/|\-|\r]+$/.test(e);
-          });
-          clearInterval(intervalObj);
-          rtm.sendMessage('```' + msg.join('') + '```', message.channel);
-        });
+    if (message.text.lastIndexOf('kontena', 0) === 0 && allowChecker(message.user)) {
+      const user = rtm.dataStore.getUserById(message.user).name
+      console.log(`${user} executed: ${message.text}`)
+      const parameters = message.text.substring('kontena'.length).trim().split(' ')
```

Comment:
IMO it's not a good idea for the bot to execute on any message that just happens to start with the word `kontena`. I mean now when I invite the bot into some channel, and someone says:
> kontena is super cool!

It would start to execute and of course fail miserably. :)

I do get the point that it was annoying to use with a DMs. 

Maybe we could check here if the message is sent as DM or on some channel and handle it bit differently? WDYT?

Tried to look for the Slack API docs how to figure out if a message is  a DM or not, no idea how it should be done. TBH the API docs are really crappy.

Reply:
I see what you mean. I've a couple of other bots for Slack and Discord, and I usually make my bot commands start with an exclamation mark. Which would also make it ideal in this case.
`!kontena node ls`

What do you think about that ?

Reply:
Also I can see we need some Regex. Cause `kontena`, `kontena-bot:` or `!kontena` any where in a sentence will trigger the bot :)

Reply:
yeah, that's why I originally preferred the more "explicit" way of getting the bots attention.